channeldb/payment_control.go

```diff
@@ -311,15 +311,12 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 		}
 
 		// We cannot register a new attempt if the payment already has
-		// reached a terminal condition. We check this before
-		// ensureInFlight because it is a more general check.
-		settle, fail := p.TerminalInfo()
-		if settle != nil || fail != nil {
-			return ErrPaymentTerminal
-		}
+		// reached a terminal condition.
+		settle, reason := p.TerminalInfo()
+		settled := settle != nil
+		failed := reason != nil
 
-		// Ensure the payment is in-flight.
-		if err := ensureInFlight(p); err != nil {
+		if err := p.Status.Registrable(settled, failed); err != nil {
```

Comment:
Shouldn't `Registrable` be a method on `p` rather than on `p.Status`, given the arguments that are passed in here?

Reply:
actually i wasn't sure since both could make sense. Switched to `p` here as it has fewer lines.